components/results_table/commons/results_table.lua

```diff
@@ -302,7 +305,132 @@ function BaseResultsTable:buildPlayersOnTeamOpponentConditions(opponentTeamTepla
 	}
 end
 
--- todo:
--- > build display from config and data
+function BaseResultsTable:build()
+	local displayTable = mw.html.create('table')
+		:addClass('wikitable wikitable-striped sortable')
+		:css('text-align', 'center')
+		:node(self:buildHeader())
+
+	if Table.isEmpty(self.data) or Table.isEmpty(self.data[1]) then
+		return displayTable:node(mw.html.create('tr')
+			:tag('td'):attr('colspan', 42):wikitext('No recorded results found.'))
+	end
+
+	for _, dataSet in ipairs(self.data) do
+		for _, row in ipairs(self:_buildRows(dataSet)) do
+			displayTable:node(row)
+		end
+	end
+
+	displayTable:node(self.args.manualContent)
+
+	return mw.html.create('div')
+		:addClass('table-responsive')
+		:node(displayTable)
+end
+
+function BaseResultsTable:_buildRows(placementData)
+	local rows = {}
+
+	if placementData.header then
+		table.insert(rows, mw.html.create('tr')
+			:tag('th'):addClass('sortbottom'):attr('colspan', 42):wikitext(placementData.header):done()
+			:done())
+	end
+
+	for _, placement in ipairs(placementData) do
+		table.insert(rows, self:buildRow(placement))
+	end
+
+	return rows
+end
+
+-- overwritable
+function BaseResultsTable:rowHighlight(placement)
+	if String.isNotEmpty(placement.publishertier) then
+		return 'valvepremier-highlighted'
+	end
+end
+
+-- overwritable
+function BaseResultsTable:tierDisplay(placement)
+	local tierDisplay
+
+	if String.isEmpty(placement.liquipediatiertype) and String.isEmpty(placement.liquipediatier) then
+		return '', ''
+	elseif String.isNotEmpty(placement.liquipediatiertype) then
+		local tierType = placement.liquipediatiertype:lower()
+		tierDisplay = Tier.text.types[tierType] or placement.liquipediatiertype
+	else
+		tierDisplay = Tier.text.tiers[placement.liquipediatier] or placement.liquipediatier
+	end
+
+	return Page.makeInternalLink(
+		{},
+		tierDisplay,
+		tierDisplay .. ' Tournaments'
+	), tierDisplay
+end
+
+-- overwritable
+function BaseResultsTable:opponentDisplay(data, options)
+	options = options or {}
+
+	if not data.opponenttype then
+		return OpponentDisplay.BlockOpponent{
+			opponent = Opponent.tbd(),
+			flip = (options or {}).flip
+		}
+	elseif (data.opponenttype ~= Opponent.solo or not options.teamForSolo)
+		and data.opponenttype ~= Opponent.team then
+
+		return OpponentDisplay.BlockOpponent{
+			opponent = Opponent.fromLpdbStruct(data),
+			flip = (options or {}).flip
+		}
+	end
+
+	local teamTemplate
+	if data.opponenttype == Opponent.team then
+		teamTemplate = data.opponenttemplate
+	else
+		teamTemplate = data.opponentplayers.p1team
+	end
+
+	if String.isEmpty(teamTemplate) then
+		return
+	end
+
+	return Team.icon(nil, teamTemplate, data.date)
```

Comment:
Add a "icon" style to OpponentDisplay's BlockTeam and use that

Reply:
will look into it later today when on pc